feat(core/modal): migrate to shadow dom (#648)
siemens/ix: BREAKING_CHANGES.md

## v2.0.0

### Modal

#### Function to display modal renamed

> This breaking change only affects the function `modal` which are exported from `@siemens/ix` (not `@siemens/ix-angular` etc).

Reviewer: ```suggestion
This breaking change only affects the function `modal` which is exported from `@siemens/ix` (not `@siemens/ix-angular` etc).
```

___Before___

```ts
export async function modal<T = any>( config: ModalConfig<T>): Promise<ModalInstance<T>>
```

___Now___
```ts
export async function showModal<T = any>( config: ModalConfig<T>): Promise<ModalInstance<T>>
```

#### Container component `ix-modal-container` removed 

> Container is not needed anymore because model system of iX is using `HTMLDialog` as

Reviewer: ```suggestion
Container is not needed anymore because the iX modal system is using `HTMLDialog` now as
```

basic system to open modal dialogs

#### Property `size` changed

___Before___

```ts
export type IxModalSize = 'sm' | 'lg' | 'xl';
```

___Now___

```ts
export type IxModalFixedSize = '360' | '480' | '600' | '720' | '840';
export type IxModalDynamicSize = 'full-width';
export type IxModalSize = IxModalFixedSize | IxModalDynamicSize;
```

#### Property ModalConfig changed/adapted

- `scrollable` is removed, no successor
- `modalDialogClass` is removed, no successor
- `windowClass` is removed, no successor
- `content` is now generic
- `backdrop = 'static'` removed, successor will be `closeOnBackdropClick`.

___Before___

```tsx
const config: ModalConfig = {
  // Other properties
  backdrop: 'static'
}
```

___Now___

```tsx
const config: ModalConfig = {
  // Other properties
  backdrop: true // `true` is default
  closeOnBackdropClick: true
}
```

> - `icon` and `iconColor` is removed, successor is `ix-modal-header` component.

Reviewer: ```suggestion
- `icon` and `iconColor` were removed, successor is the `ix-modal-header` component.
```

#### Property `backdropClass` removed

> To get a consistent ui design be decided to remove custom backdrops.

Reviewer: ```suggestion
To get a consistent UI design we decided to remove custom backdrops.
```
